app/controllers/application_controller.rb

```diff
@@ -12,12 +14,46 @@ class ApplicationController < ActionController::Base
 
   # Determine if the collection is viewable depending on private flag and white-listed remote IP address
   def is_viewable? (collection)
-    !collection.is_private || (collection.allowed_ip_addresses.split(%r{,\s*}).include? request.remote_ip)
+
+    # If no ip address filtering
+    if (collection['allowed_ip_addresses'].empty?)
+      if (is_private?(collection))
+        # TODO: Brittle - Allow better control over who may view a private collecionts
+        RoleMapper.roles(current_user.email).include? "archivist"
```

Comment:
Email is passed to RoleMapper.roles method to get the Role mapping in the config/role_map.yml. current_user is the user that's logged into the app by way of http://locahost:3000/users/sign_in.

- Note - I updated the PR description with an additional item to discuss.